examples/fcis/train.py

```diff
@@ -0,0 +1,183 @@
+    args = parser.parse_args()
+
+    np.random.seed(args.seed)
+
+    # dataset
+    train_dataset = SBDInstanceSegmentationDataset(split='train')
+    test_dataset = SBDInstanceSegmentationDataset(split='val')
+
+    # model
+    fcis = FCISResNet101(
+        n_fg_class=len(sbd_instance_segmentation_label_names),
+        pretrained_model='imagenet', iter2=False)
+    fcis.use_preset('evaluate')
+    model = FCISTrainChain(fcis)
+
+    # gpu
+    if args.gpu >= 0:
+        chainer.cuda.get_device_from_id(args.gpu).use()
+        model.to_gpu()
+
+    # optimizer
+    optimizer = chainer.optimizers.MomentumSGD(lr=args.lr, momentum=0.9)
+    optimizer.setup(model)
+
+    model.fcis.head.conv1.W.update_rule = MomentumSGDRule(
+        lr=args.lr * 3, momentum=0.9)
+    model.fcis.head.conv1.b.update_rule = MomentumSGDRule(
+        lr=args.lr * 3, momentum=0.9)
+    optimizer.add_hook(chainer.optimizer.WeightDecay(rate=0.0005))
+
+    for param in model.params():
+        if param.name in ['beta', 'gamma']:
+            param.update_rule.enabled = False
+    model.fcis.extractor.conv1.disable_update()
+    model.fcis.extractor.res2.disable_update()
+
+    train_dataset = TransformDataset(
+        train_dataset, Transform(model.fcis))
+
+    # iterator
+    train_iter = chainer.iterators.SerialIterator(
+        train_dataset, batch_size=1)
+    test_iter = chainer.iterators.SerialIterator(
+        test_dataset, batch_size=1, repeat=False, shuffle=False)
+    updater = chainer.training.updater.StandardUpdater(
+        train_iter, optimizer, converter=concat_examples,
+        device=args.gpu)
+
+    trainer = chainer.training.Trainer(
+        updater, (args.epoch, 'epoch'), out=args.out)
+
+    # lr scheduler
+    trainer.extend(
+        chainer.training.extensions.ExponentialShift(
+            'lr', args.lr_cooldown_factor, init=args.lr),
+        trigger=(args.cooldown_epoch, 'epoch'))
+
+    # interval
+    log_interval = 100, 'iteration'
+    plot_interval = 3000, 'iteration'
+    print_interval = 20, 'iteration'
+
+    # training extensions
+    trainer.extend(
+        extensions.snapshot_object(
+            model.fcis, filename='snapshot_model.npz'),
+        trigger=(args.epoch, 'epoch'))
+    trainer.extend(
+        extensions.observe_lr(),
+        trigger=log_interval)
+    trainer.extend(
+        extensions.LogReport(log_name='log.json', trigger=log_interval))
+    trainer.extend(extensions.PrintReport([
+        'iteration', 'epoch', 'elapsed_time', 'lr',
+        'main/loss',
+        'main/rpn_loc_loss',
+        'main/rpn_cls_loss',
+        'main/roi_loc_loss',
+        'main/roi_cls_loss',
+        'main/roi_mask_loss',
+        'validation/main/map',
+    ]), trigger=print_interval)
+    trainer.extend(
+        extensions.ProgressBar(update_interval=10))
+
+    if extensions.PlotReport.available():
+        import matplotlib
+        matplotlib.use('Agg')
```

Comment:
Can you not set matplotlib setting inside a script?

(related, look at the end of this section 
 https://github.com/chainer/chainercv/tree/master/examples/faster_rcnn#train-code)